### PR TITLE
UOE-10197: Support Adunit config for AMP(in-banner-video)

### PR DESCRIPTION
**Pull request by AvinashKapre**

### CODE REVIEW

File: modules/pubmatic/openwrap/util.go [hunk 1]
```diff
@@ -310,3 +312,18 @@ func getPlatformFromRequest(request *openrtb2.BidRequest) string {
...
 }
+
+// for AMP requests based on traffic percentage, we will decide to send video or not
+// if traffic percentage is not defined then send video
+// if traffic percentage is defined then send video based on percentage
+func checkIsVideoEnabledForAMP(endpont string, adUnitConfig *adunitconfig.AdConfig) bool {
+	if adUnitConfig == nil || adUnitConfig.Video == nil || adUnitConfig.Video.Enabled == nil || !*adUnitConfig.Video.Enabled {
+		return false
+	}
+	if endpont == models.EndpointAMP {
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/util.go:323]: The method name suggests we are doing this for the AMP endpoint only, can we move this check outside the method call?

File: modules/pubmatic/openwrap/util.go [hunk 2]
```diff
@@ -310,3 +312,18 @@ func getPlatformFromRequest(request *openrtb2.BidRequest) string {
 	}
 	return platform
 }
+
+// for AMP requests based on traffic percentage, we will decide to send video or not
+// if traffic percentage is not defined then send video
+// if traffic percentage is defined then send video based on percentage
+func checkIsVideoEnabledForAMP(endpont string, adUnitConfig *adunitconfig.AdConfig) bool {
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/util.go:319]: Nitpick: rename endpont to endpoint

File: modules/pubmatic/openwrap/beforevalidationhook.go [hunk 3]
```diff
@@ -991,3 +924,199 @@ func getTagID(imp openrtb2.Imp, impExt *models.ImpExtension) string {
...
+	}
+	if imp.Video.SkipMin == 0 {
+		imp.Video.SkipMin = 0
+	}
+	if imp.Video.SkipAfter == 0 {
+		imp.Video.SkipAfter = 0
+	}
+	if imp.Video.BoxingAllowed == 0 {
+		imp.Video.BoxingAllowed = 0
+	}
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/beforevalidationhook.go:1084]: Do we need this if values are already set to the desired ones?

File: modules/pubmatic/openwrap/util_test.go [hunk 4]
```diff
@@ -1076,3 +1078,91 @@ func Test_getIP(t *testing.T) {
 		})
 	}
 }
+
+func Test_checkIsVideoEnabledForAMP(t *testing.T) {
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/util_test.go:1082]: Nitpick: Use camelcase

File: modules/pubmatic/openwrap/util.go (hunk 2 above)
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/util.go:319]: Rename the method to `isVideoEnabledForAMP`

File: modules/pubmatic/openwrap/beforevalidationhook.go (hunk 3 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/beforevalidationhook.go:1084]: Yes, we can remove this.

File: modules/pubmatic/openwrap/util.go (hunk 1 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/util.go]: Done

File: modules/pubmatic/openwrap/util.go (hunk 2 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/util.go]: Done

File: modules/pubmatic/openwrap/beforevalidationhook.go (hunk 3 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/beforevalidationhook.go]: Done

File: modules/pubmatic/openwrap/util.go (hunk 2 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/util.go]: Done

**Review by ashishshinde-pubm — APPROVED** [review]
LGTM

**Review by pm-isha-bharti — APPROVED** [review]

**Pull request merged**